doc/nrf/libraries/others/event_manager_profiler.rst

```diff
@@ -0,0 +1,109 @@
+.. _event_manager_profiler:
+
+Event Manager Profiler
+######################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Event Manager Profiler is a piece of software that supports profiling for Event Manager.
+It serves as a linking layer between :ref`event_manager` and :ref:`profiler`.
+Together with Profiler, Event Manager Profiler enables logging and visualizing data for performance measurements, while the system is running.
+
+
+Configuration
+*************
+
+To use Event Manager Profiler, you must enable the following Kconfig option:
+
+* :kconfig:`CONFIG_EVENT_MANAGER_PROFILER` - This option enables the Event Manager Profiler.
+
+
+Implementation
+**************
+
+Event Manager Events are structured data types that are defined by the application and can contain additional data.
+The Event Manager Profiler expands each events data structure by adding to it profiler_info structure which holds information required by Profiler.
+Using those extra information Event Manager Profiler tracks usage of events.
+If EVENT_MANAGER_PROFILER_TRACE_EVENT_EXECUTION option is set Event Manager Profiler will track two extra events which mark start and stop of each events execution.
+
+
+Implementing profiling for Event Manager Events
+***********************************************
+
+If an application module is supposed to react to an event, your application must implement an event type, submit the event, and register the module as listener.
+Read the following sections for details.
+
+
+Profiling an event
+==================
+
+The Event Manager events can be profiled using :ref:`profiler`.
+Profiler allows you to observe the propagation of an event in the system, view the data connected with the event, or create statistics.
+
+To profile an event, you must complete the following steps:
+
+1. Enable the profiler using the :kconfig:`CONFIG_EVENT_MANAGER_PROFILER` Kconfig option.
+#. Edit the source file for the event type:
+
+   a. Define a profiling function that logs the event data to a given buffer by calling one of the following functions for every registered data type:
+
+      * :c:func:`profiler_log_encode_uint32`
+      * :c:func:`profiler_log_encode_int32`
+      * :c:func:`profiler_log_encode_uint16`
+      * :c:func:`profiler_log_encode_int16`
+      * :c:func:`profiler_log_encode_uint8`
+      * :c:func:`profiler_log_encode_int8`
+      * :c:func:`profiler_log_encode_string`
+   #. Define an :c:struct:`profiler_info` structure, using :c:macro:`EVENT_INFO_DEFINE` in your event source file, and provide it as an argument when defining the event type with :c:macro:`EVENT_TYPE_DEFINE` macro.
```

Comment:
provide it as trace data ?

Reply:
Maybe we could actually describe the tracing hooks provided by the Event Manager and then rely on the description to describe profiler integration?
The hooks are quire generic and they could be used for various purposes, so they may be worth documenting

Reply:
OK, I will add their descriptions.